source/includes/fundamentals/code-examples/Cursor.cs

```diff
@@ -0,0 +1,155 @@
+                    {
+                        Console.WriteLine(restaurant.Name);
+                    }
+                }
+            }
+            // end-cursor-iterate
+        }
+
+        {
+            // start-cursor-iterate-async
+            var filter = Builders<Restaurant>.Filter.Empty;
+            
+            using (var cursor = await collection.FindAsync(filter))
+            {
+                while (await cursor.MoveNextAsync())
+                {
+                    foreach (var restaurant in cursor.Current)
+                    {
+                        Console.WriteLine(restaurant.Name);
+                    }
+                }
+            }
+            // end-cursor-iterate-async
+        }
+
+        {
+            // start-cursor-iterate-to-cursor
+            var filter = Builders<Restaurant>.Filter.Empty;
+            
+            using (var cursor = collection.Find(filter).ToCursor())
+            {
+                while (cursor.MoveNext())
+                {
+                    foreach (var restaurant in cursor.Current)
+                    {
+                        Console.WriteLine(restaurant.Name);
+                    }
+                }
+            }
+            // end-cursor-iterate-to-cursor
+        }
+
+        {
+            // start-cursor-iterate-to-cursor-async
+            var filter = Builders<Restaurant>.Filter.Empty;
+            
+            using (var cursor = await collection.Find(filter).ToCursorAsync())
+            {
+                while (await cursor.MoveNextAsync())
+                {
+                    foreach (var restaurant in cursor.Current)
+                    {
+                        Console.WriteLine(restaurant.Name);
+                    }
+                }
+            }
+            // end-cursor-iterate-to-cursor-async
+        }
+
+        {
+            // start-cursor-to-list
+            var filter = Builders<Restaurant>.Filter.Eq(r => r.Name, "Dunkin' Donuts");
+            var results = collection.FindSync(filter).ToList();
+            // end-cursor-to-list
+        }
+
+        {
+            // start-cursor-to-list-async
+            var filter = Builders<Restaurant>.Filter.Eq(r => r.Name, "Dunkin' Donuts");
+            var results = (await collection.FindAsync(filter)).ToList();
+            // end-cursor-to-list-async
+        }
+
+        {
+            // start-tailable-cursor
+            var filter = Builders<Restaurant>.Filter.Eq(r => r.Name, "Dunkin' Donuts");
+            var options = new FindOptions<Restaurant>
+            {
+                CursorType = CursorType.TailableAwait
+            };
+
+            using (var cursor = collection.FindSync(filter, options))
+            {
+                while (cursor.MoveNext())
+                {
+                    foreach (var restaurant in cursor.Current)
+                    {
+                        // Process each restaurant
```

Comment:
S: as in the other examples, show printing the current restaurant (same for async example)